vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SlideManagerTest.java

```diff
@@ -0,0 +1,124 @@
+        moduleId = "MOD000000002";
+        sequenceId = "SEQ000000004";
+        slidesList.get(0).setId(slideId);
+    }    
+ 
+    @Test
+    public void test_slideSequence_slideExistsInSequence() {
+        
+        // Positive scenario - Slide present in Sequence
+        
+        sequenceObj.setId(sequenceId);
+        sequenceObj.setSlides(slidesList);
+        List<Sequence> sequencesList = new ArrayList<>();
+        sequencesList.add(sequenceObj);
+        Mockito.when(sequenceRepo.findSequencesForModule(moduleId)).thenReturn(sequencesList);
+        List<Sequence> slideSequencePresent = slideManagerToTest.getSlideSequences(slideId, moduleId);
+    
+        String actualSlideIdSequence = slideSequencePresent.get(0).getSlides().get(0).getId();
+        String expectedSlideIdSequence = sequencesList.get(0).getSlides().get(0).getId();
+        
+        Assert.assertEquals(actualSlideIdSequence, expectedSlideIdSequence);
+        Assert.assertEquals(slideSequencePresent.size(), sequencesList.size()); 
+        
+    }
+    
+    @Test
+    public void test_slideSequence_slideNotExistsInSequence() {
+    	
+    	List<Sequence> slideSequencePresent = slideManagerToTest.getSlideSequences(slideId, moduleId);
+        List<Sequence> actualSequenceSlideListNotPresent = slideManagerToTest.getSlideSequences(slideId, moduleId);
+        Assert.assertNotSame(actualSequenceSlideListNotPresent, slideSequencePresent);
+    }
+    
+    @Test
+    public void test_deleteSlideById_slideIdPresent() throws SlideDoesNotExistException {
+        
+        
+        sequenceObj.setId(sequenceId);
+        sequenceObj.setSlides(slidesList);
+        List<Sequence> sequencesList = new ArrayList<>();
+        sequencesList.add(sequenceObj);
+        Mockito.when(sequenceRepo.findSequencesForModule(moduleId)).thenReturn(sequencesList);
+        
+        // using mockito spy object
+        ISlide slide = new Slide();
+        slide.setId(slideId);
+        Mockito.doReturn(slide).when(slideManagerToTestSpy).getSlide(Mockito.any());
+        
+        slideManagerToTestSpy.deleteSlideById(slideId, moduleId);  
+        Mockito.verify(slideRepo).delete((Slide) slide);
+    }
+    
+    @Test
+    public void test_deleteSlideById_slideIdNotPresent() throws SlideDoesNotExistException {
+    	
+    	
+    	Mockito.when(slideRepo.findById(Mockito.anyString())).thenReturn(Optional.empty());
+        Assert.assertNull(slideManagerToTest.getSlide(Mockito.anyString()));
+    	
+    	sequenceObj.setId(sequenceId);
+        sequenceObj.setSlides(slidesList);
+        List<Sequence> sequencesList = new ArrayList<>();
+        sequencesList.add(sequenceObj);
+        Mockito.when(sequenceRepo.findSequencesForModule(moduleId)).thenReturn(sequencesList);
+        
+        // using mockito spy object
+        ISlide slide = new Slide();
+        slide.setId(slideIdNotPresent);
+        Mockito.doReturn(slide).when(slideManagerToTestSpy).getSlide(Mockito.any());

```

Comment:
same as above, why a spy?

Reply:
If I use the actualy SlideManager object, that is slideManagerToTest and then call it's class defined methods, while debugging it shows not a mock object and tries to hit actual DB, thereby returning null.

Hence I am using a spy object.